File: 1.two-sum/golang/tests/two_sum_test.go
package tests

import (
	"github.com/stretchr/testify/assert"
	"testing"
	"two-sum/src"
)

type Inputs struct {
	nums   []int
	target int
}

func TestCase01(t *testing.T) {
	inputs := Inputs{
		nums:   []int{2, 7, 11, 15},
		target: 9,
	}
	expected := []int{0, 1}

	processTestCase(t, inputs, expected)
}

func TestCase02(t *testing.T) {
	inputs := Inputs{
		nums:   []int{3, 2, 4},
		target: 6,
	}
	expected := []int{1, 2}

	processTestCase(t, inputs, expected)
}

func TestCase03(t *testing.T) {
	inputs := Inputs{
		nums:   []int{3, 3},
		target: 6,
	}
	expected := []int{0, 1}

	processTestCase(t, inputs, expected)
}

func processTestCase(t *testing.T, inputs Inputs, expected []int) {
	actual := solution.TwoSum(inputs.nums, inputs.target)
	assert.Equal(t, expected, actual)
}

File: 1.two-sum/java/src/test/java/company/app/TwoSumTest.java
package company.app;

import org.junit.Test;
import static org.junit.Assert.assertArrayEquals;

/**
 * Unit tests
 */
public class TwoSumTest
{
    TwoSum twoSum = new TwoSum();

    @Test
    public void testCase01()
    {
        Inputs inputs = new Inputs();
        inputs.setNums(new int[] {2, 7, 11, 15})
              .setTarget(9);
      
        int[] expected = {0, 1};

        processTestCase(inputs, expected);
    }

    @Test
    public void testCase02()
    {
      Inputs inputs = new Inputs();
      inputs.setNums(new int[] {3, 2, 4})
            .setTarget(6);
      
      int[] expected = {1, 2};

      processTestCase(inputs, expected);
    }

    @Test
    public void testCase03()
    {
      Inputs inputs = new Inputs();
      inputs.setNums(new int[] {3, 3})
            .setTarget(6);

      int[] expected = {0, 1};

      processTestCase(inputs, expected);
    }

    public void processTestCase(Inputs inputs, int[] expected)
    {
        int[] actual = twoSum._do(inputs.getNums(), inputs.getTarget());
        assertArrayEquals(expected, actual);
    }

    private class Inputs
    {
        private int[] nums;
        private int target;

        public int[] getNums()
        {
          return this.nums;
        }

        public int getTarget()
        {
          return this.target;
        }

        public Inputs setNums(int[] nums)
        {
          this.nums = nums;
          return this;
        }

        public Inputs setTarget(int target)
        {
          this.target = target;
          return this;
        }
    }
}
